### ISSUE TITLE
Emacs 31.1 is still treated as unreleased

### ISSUE THREAD

**Issue by Thysrael** [purcell/package-lint#312]
### Summary

package-lint 0.26 and the current default branch still define `package-lint-emacs-head-version` as `(31)`. Emacs 31.1 was released on 2026-08-24, so packages that correctly require Emacs 31.1 now receive a false warning.

Official release announcement: https://lists.gnu.org/archive/html/emacs-devel/2026-08/msg00760.html

### Reproduction

Lint a package containing:

```elisp
;; Package-Requires: ((emacs "31.1"))
```

Current result:

```text
warning: This makes the package uninstallable in all released Emacs versions.
```

I reproduced this with package-lint commit `87bf02ca387a37094e1a0057adefa9735d880cec`.

### Expected result

No warning for Emacs 31.1. Emacs HEAD is now version 32.

This appears equivalent to the update made in #292: bump `package-lint-emacs-head-version` from `(31)` to `(32)` and adjust `package-lint-test-warning-emacs-head-dep` to use Emacs 32.
